//generator:protoc-gen-js does not depend on a module exporting 'google/protobuf/compiler/plugin.h'

The following error occurs when building `//generator:protoc-gen-js` with Bazel:

```
❯ bazel build //generator:protoc-gen-js
INFO: Analyzed target //generator:protoc-gen-js (0 packages loaded, 0 targets configured).
ERROR: /Users/luangong/Documents/protobuf/protobuf-javascript/generator/BUILD.bazel:1:10: Compiling generator/protoc-gen-js.cc failed: (Exit 1): cc_wrapper.sh failed: error executing CppCompile command (from target //generator:protoc-gen-js) external/local_config_cc/cc_wrapper.sh -U_FORTIFY_SOURCE -fstack-protector -Wall -Wthread-safety -Wself-assign -Wunused-but-set-parameter -Wno-free-nonheap-object -fcolor-diagnostics ... (remaining 151 arguments skipped)

Use --sandbox_debug to see verbose messages from the sandbox and retain the sandbox build root for debugging
generator/protoc-gen-js.cc:31:10: error: module //generator:protoc-gen-js does not depend on a module exporting 'google/protobuf/compiler/plugin.h'
#include <google/protobuf/compiler/plugin.h>
         ^
1 error generated.
Target //generator:protoc-gen-js failed to build
Use --verbose_failures to see the command lines of failed build steps.
INFO: Elapsed time: 1.258s, Critical Path: 1.15s
INFO: 5 processes: 5 internal.
ERROR: Build did NOT complete successfully
```

--- a/generator/BUILD.bazel
+++ b/generator/BUILD.bazel
@@ -10,7 +10,14 @@ cc_binary(
     visibility = ["//visibility:public"],
     deps = [
         "@com_google_protobuf//:protobuf",
+        "@com_google_protobuf//src/google/protobuf/compiler:code_generator",
+        "@com_google_protobuf//src/google/protobuf/io",
+        "@com_google_protobuf//src/google/protobuf/io:printer",
+        "@com_google_protobuf//src/google/protobuf/io:tokenizer",
         "@com_google_protobuf//:protoc_lib",
+        "@com_google_absl//absl/strings",
+        "@com_google_absl//absl/strings:str_format",
     ],
 )
+
 

--- a/generator/js_generator.cc
+++ b/generator/js_generator.cc
@@ -36,8 +36,6 @@
 #include <google/protobuf/io/printer.h>
 #include <google/protobuf/io/strtod.h>
 #include <google/protobuf/io/zero_copy_stream.h>
-#include <google/protobuf/stubs/common.h>
-#include <google/protobuf/descriptor_legacy.h>
 #include "absl/strings/escaping.h"
 #include "absl/strings/str_cat.h"
 #include "absl/strings/str_format.h"

--- a/generator/js_generator.h
+++ b/generator/js_generator.h
@@ -36,11 +36,8 @@
 #include <set>
 #include <string>
 
-#include <google/protobuf/stubs/common.h>
 #include <google/protobuf/compiler/scc.h>
 #include <google/protobuf/compiler/code_generator.h>
-
-#include <google/protobuf/port_def.inc>
 
 namespace google {
 namespace protobuf {
@@ -133,7 +130,7 @@ struct GeneratorOptions {
 // header.  If you create your own protocol compiler binary and you want it to
 // support JavaScript output, you can do so by registering an instance of this
 // CodeGenerator with the CommandLineInterface in your main() function.
-class PROTOC_EXPORT Generator : public CodeGenerator {
+class Generator : public CodeGenerator {
  public:
   Generator() {}
   virtual ~Generator() {}
@@ -330,7 +327,5 @@ class PROTOC_EXPORT Generator : public CodeGenerator {
 }  // namespace compiler
 }  // namespace protobuf
 }  // namespace google
-
-#include <google/protobuf/port_undef.inc>
 
 #endif  // GOOGLE_PROTOBUF_COMPILER_JS_GENERATOR_H__

